config/adapter.go

```diff
@@ -1,9 +1,14 @@
 package config
 
+import "github.com/prebid/prebid-server/util/randomutil"
+
 type Adapter struct {
 	Endpoint         string
 	ExtraAdapterInfo string
 
+	// needed for AppNexus
+	RandomGenerator randomutil.RandomGenerator
```

Comment:
Please do not add this field to the Adapter struct. To fake out a field used for test, follow the pattern from the adnuntius adapter where you cast the adapter after the builder:

```
func TestJsonSamples(t *testing.T) {
	bidder, buildErr := Builder(openrtb_ext.BidderAppnexus, config.Adapter{
		Endpoint: "http://ib.adnxs.com/openrtb2"}, config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})

	if buildErr != nil {
		t.Fatalf("Builder returned unexpected error %v", buildErr)
	}
	
        bidderAppNexus, _ := bidder.(*adapter)
	bidderAppNexus.randomGenerator = <fake one>

	adapterstest.RunJSONBidderTest(t, "appnexustest", bidder)
}
```

Reply:
done in https://github.com/prebid/prebid-server/pull/2679/commits/868dbdbbdace65b645bb5e28ccb5df68aadbfa9b